packages/itwin/tree-widget/src/components/trees/models-tree/HierarchyBasedVisibilityHandler.ts

```diff
@@ -547,6 +541,41 @@ class VisibilityHandlerImplementation implements IVisibilityHandler {
     return ovr ? from(ovr(this.createVoidOverrideProps(props, result))) : result;
   }
 
+  private showModelWithoutAnyCategoriesOrElements(modelId: Id64String) {
+    const viewport = this._props.viewport;
+    return forkJoin({
+      categories: this._queryHandler.queryModelCategories(modelId).pipe(toArray()),
+      alwaysDrawnChildren: this.getAlwaysDrawnChildren({ modelId }),
+      _: viewport.addViewedModels(modelId),
```

Comment:
I think we should first disable display of categories & elements and only then enable display of the model. As it is now, for a moment we'll see all model content, and then it's going to be hidden again.

Reply:
Fixed